lib/internal/bootstrap_node.js

```diff
@@ -108,6 +108,28 @@
         'ExperimentalWarning', undefined);
     }
 
+    {
+      // Install legacy getters on the `util` binding for typechecking.
+      // TODO(addaleax): Turn into a full runtime deprecation.
+      const { pendingDeprecation } = process.binding('config');
+      const { deprecate } = NativeModule.require('internal/util');
+      const utilBinding = process.binding('util');
+      const types = internalBinding('types');
+      for (const name of [
+        'isArrayBuffer', 'isArrayBufferView', 'isAsyncFunction',
+        'isDataView', 'isDate', 'isExternal', 'isMap', 'isMapIterator',
+        'isNativeError', 'isPromise', 'isRegExp', 'isSet', 'isSetIterator',
+        'isTypedArray', 'isUint8Array', 'isAnyArrayBuffer'
+      ]) {
+        utilBinding[name] = pendingDeprecation ?
+          deprecate(types[name],
+                    'Accessing native typechecking bindings of Node ' +
+                    'directly is deprecated. ' +
+                    `Please use \`util.types.${name}\` instead.`,
+                    'DEP0XXX') :
```

Comment:
If a new deprecation is introduced, it should be recorded in [doc/api/deprecations.md](https://github.com/nodejs/node/blob/master/doc/api/deprecations.md).